bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/providers/SonyDefinitionProviderImpl.java

```diff
@@ -0,0 +1,384 @@
+                List<StateOption> options = null;
+
+                // First use any specified override (if found)
+                // Note since compiler thinks overrideDesc cannot be null
+                // it flags the 'readonly' below as can't be null (which is incorrect)
+                final StateDescription overrideDesc = stateOverride.get(id);
+                if (overrideDesc != null) {
+                    min = overrideDesc.getMinimum();
+                    max = overrideDesc.getMaximum();
+                    step = overrideDesc.getStep();
+                    pattern = overrideDesc.getPattern();
+                    readonly = overrideDesc.isReadOnly();
+                    options = overrideDesc.getOptions();
+                }
+
+                // Finally use the original values
+                if (originalStateDescription != null) {
+                    if (min == null) {
+                        min = originalStateDescription.getMinimum();
+                    }
+
+                    if (max == null) {
+                        max = originalStateDescription.getMaximum();
+                    }
+
+                    if (step == null) {
+                        step = originalStateDescription.getStep();
+                    }
+
+                    if (pattern == null) {
+                        pattern = originalStateDescription.getPattern();
+                    }
+
+                    if (readonly == null) {
+                        readonly = originalStateDescription.isReadOnly();
+                    }
+
+                    if (options == null) {
+                        options = originalStateDescription.getOptions();
+                    }
+                }
+
+                // If anything is specified, create a new state description and go with it
+                if (min != null || max != null || step != null || pattern != null || readonly != null
+                        || (options != null && !options.isEmpty())) {
+                    StateDescriptionFragmentBuilder bld = StateDescriptionFragmentBuilder.create();
+                    if (min != null) {
+                        bld = bld.withMinimum(min);
+                    }
+                    if (max != null) {
+                        bld = bld.withMaximum(max);
+                    }
+                    if (step != null) {
+                        bld = bld.withStep(step);
+                    }
+                    if (pattern != null) {
+                        bld = bld.withPattern(pattern);
+                    }
+                    if (readonly != null) {
+                        bld = bld.withReadOnly(readonly);
+                    }
+                    if (!options.isEmpty()) {
+                        bld = bld.withOptions(options);
+                    }
+                    return bld.build().toStateDescription();
+                }
+            }
+        }
+        return originalStateDescription;
```

Comment:
@cweitkamp @tmrobert8 Should null also be returned here as well, since this method is called by line 186?

Reply:
lol - already did that - look at the commit above

Reply:
Yep, you're fast. Thanks!!